### ISSUE TITLE
Why is the precision of deepshift higher than that of multiplication in some experiments？

### ISSUE THREAD

**Issue by mengjingyouling** [mostafaelhoushi/DeepShift#20]
hi @mostafaelhoushi 

We have done some experiments. As reported in your paper, the precision of shift in some experiments is higher than that of multiplication.
Have you ever thought about the reason? We think that it is because of the sparsity（discontinuous） realized by the Deepshift that the model can converge to the optimal solution instead of the suboptimal solution.

Thanks

**Comment by mostafaelhoushi** [comment]
Hi @mengjingyouling 
Happy to hear that result is higher than multiplication. 
I think this is reported in other quantization papers as well such as [INQ](https://openreview.net/pdf?id=HyQJ-mclg) and in some pruning papers as well.
It could be because that discontinuous effect as you described is regularizing: adds some error to training set but reduces overfitting and hence increases accuracy on validation set.
This paper discusses that: https://arxiv.org/abs/2206.12372

**Comment by mengjingyouling** [comment]
I get it,thank you.
